restli-common/src/main/java/com/linkedin/restli/common/util/ProjectionMaskApplier.java

```diff
@@ -215,6 +215,11 @@ else if (wildcardMask != null)
 
     UnionDataSchema newUnionDataSchema = new UnionDataSchema();
     newUnionDataSchema.setMembers(newUnionMembers, errorMessageBuilder);
+    if (newUnionMembers.size() > 0 && unionDataSchema.getMembers().size() > newUnionMembers.size())
```

Comment:
why do you need size > 0 check?

newUnionmember size will be zero only if the mask is empty. Which means none of the members were projected. We should treat that as partial projection.

It is not possible to create that projection using pathspecs, but someone building the mask map directly can do it.

Reply:
For the last sentence, is the following code kind of using pathspecs to create projection?

```
    List<PathSpec> spec = Arrays.asList(
        ValidateEmptyUnion.fields().foo().Fuzz(),
        ValidateEmptyUnion.fields().foo().Bar());

```

Reply:
I meant creating an empty projection for a union., with the following mask map:
{ 
   foo: {}
}

using ValidateEmptyUnion.fields().foo() will result in:
{
  foo: 1
 }
 And ValidateEmptyUnion.fields().foo().Fuzz() will result in:
 {
  foo:  {
    Fuzz: 1
  }
 }